src/TransparentValueObjects/ValueObjectIncrementalSourceGenerator.cs

```diff
@@ -346,6 +352,39 @@ string TypeSymbolToString(ITypeSymbol typeSymbol)
         cw.AppendLine();
     }
 
+    public static void AddSystemTextJsonClasses(CodeWriter cw, string valueObjectTypeName, string innerValueTypeName)
+    {
+        cw.AppendLine($"public class SystemTextJsonConverter : global::System.Text.Json.Serialization.JsonConverter<{valueObjectTypeName}>");
+        using (cw.AddBlock())
+        {
+            cw.AppendLine($"public override {valueObjectTypeName} Read(ref global::System.Text.Json.Utf8JsonReader reader, global::System.Type typeToConvert, global::System.Text.Json.JsonSerializerOptions options)");
+            using (cw.AddBlock())
+            {
+                cw.AppendLine($"var valueObject = global::System.Text.Json.JsonSerializer.Deserialize<{innerValueTypeName}>(ref reader, options);");
+                cw.AppendLine($"return {valueObjectTypeName}.From(valueObject);");
+            }
+
+            cw.AppendLine($"public override void Write(global::System.Text.Json.Utf8JsonWriter writer, {valueObjectTypeName} value, global::System.Text.Json.JsonSerializerOptions options)");
+            using (cw.AddBlock())
+            {
+                cw.AppendLine("global::System.Text.Json.JsonSerializer.Serialize(writer, value.Value, options);");
+            }
+
+            cw.AppendLine($"public override {valueObjectTypeName} ReadAsPropertyName(ref global::System.Text.Json.Utf8JsonReader reader, global::System.Type typeToConvert, global::System.Text.Json.JsonSerializerOptions options)");
+            using (cw.AddBlock())
+            {
+                cw.AppendLine($"var valueObject = global::System.Text.Json.JsonSerializer.Deserialize<{innerValueTypeName}>(ref reader, options);");
+                cw.AppendLine($"return {valueObjectTypeName}.From(valueObject);");
+            }
+
+            cw.AppendLine($"public override void WriteAsPropertyName(global::System.Text.Json.Utf8JsonWriter writer, {valueObjectTypeName} value, global::System.Text.Json.JsonSerializerOptions options)");
+            using (cw.AddBlock())
+            {
+                cw.AppendLine("writer.WritePropertyName(global::System.Text.Json.JsonSerializer.Serialize(value.Value, options));");
+            }
+        }
+    }
```

Comment:
We can get a converter for the inner value using the `JsonSerializerOptions`:

```csharp
public class JsonConverter : JsonConverter<SampleValueObject>
{
    private static JsonConverter<string> GetInnerValueConverter(JsonSerializerOptions options)
    {
        var innerValueConverter = options.GetConverter(typeof(string));
        if (innerValueConverter is null) throw new JsonException($"Unable to find converter for type {typeof(string)}");
        return (JsonConverter<string>)innerValueConverter;
    }

    public override SampleValueObject Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
    {
        var innerValueConverter = GetInnerValueConverter(options);
        var innerValue = innerValueConverter.Read(ref reader, typeof(string), options);
        return innerValue is null ? DefaultValue : From(innerValue);
    }

    public override void Write(Utf8JsonWriter writer, SampleValueObject value, JsonSerializerOptions options)
    {
        var innerValueConverter = GetInnerValueConverter(options);
        innerValueConverter.Write(writer, value.Value, options);
    }

    public override SampleValueObject ReadAsPropertyName(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
    {
        var innerValueConverter = GetInnerValueConverter(options);
        var innerValue = innerValueConverter.ReadAsPropertyName(ref reader, typeof(string), options);
        return From(innerValue);
    }

    public override void WriteAsPropertyName(Utf8JsonWriter writer, SampleValueObject value, JsonSerializerOptions options)
    {
        var innerValueConverter = GetInnerValueConverter(options);
        innerValueConverter.WriteAsPropertyName(writer, value.Value, options);
    }
}
```

Reply:
Replaced `is null` with `== default`